### PR TITLE
SOLR-15367 Convert "rid" functionality into a default Tracer

### PR DESCRIPTION
**Pull request by stillalex**
https://issues.apache.org/jira/browse/SOLR-15367

# Description

This PR is for deprecating the rid generation, sourcing it from the trace id if that exists.
Params have been marked as deprecated with 9.4.
I have moved the changes.txt entry to 9.4 features.

This step is setup for the 9.x backport mainly. There will be a followup for main branch to remove rid completely.

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/solr-ref-guide/modules/deployment-guide/pages/distributed-tracing.adoc [hunk 1]
```diff
@@ -25,6 +25,17 @@ A sampled distributed tracing query request displayed in Jaeger UI looks like th
...
+
+Starting from 9.4 Solr ships with an always-on trace id generator. This will inject trace id headers for every Solr request, propagating the client supplied value or generating a new id as needed and replaces the existing `rid` mechanism.
+
+The header name it uses for propagation is `X-Trace-Id`, this can be changed by updating the system property `solr.traceIdHeader`.
+
+This plugin will not be enabled if another Tracing mechanism is setup via Solr config.
+This plugin can be disabled by setting the `solr.alwaysOnTraceId` system property to `false`.
+
+Note: Solr 9.4 ships with an `OpenTracing` based lightweight `Tracer` implementation and Solr 10 has an updated `OpenTelemetry` based `TextMapPropagator` implementation.
+
```
Comment (dsmiley) [inline solr/solr-ref-guide/modules/deployment-guide/pages/distributed-tracing.adoc:38]: @ctargett at some point did you say something about reference guide style in which we shouldn't refer to older or newer versions?  After all, the ref guide is itself versioned.  I don't think you said exactly that but I recall you said something similar.

File: solr/solrj/src/java/org/apache/solr/common/params/CommonParams.java [hunk 2]
```diff
@@ -277,9 +279,23 @@ public static EchoParamStyle get(String v) {
...
+  /**
+   * An opt-out flag to prevent the addition of {@link #REQUEST_ID} tracing on distributed queries
+   *
+   * <p>Defaults to 'false' if not specified.
+   *
+   * @see #DISABLE_REQUEST_ID
+   * @deprecated this was replaced by the auto-generated trace ids
+   */
+  @Deprecated(since = "9.4")
+  boolean DISABLE_REQUEST_ID_DEFAULT = Boolean.getBoolean("solr.disableRequestId");
```
Comment (dsmiley) [inline solr/solrj/src/java/org/apache/solr/common/params/CommonParams.java:297]: Admittedly I hate seeing values in the Params classes but this practice started long ago and I see you are attempting to match some loosely followed conventions.

**Review by dsmiley — APPROVED** [review]

File: solr/solr-ref-guide/modules/deployment-guide/pages/distributed-tracing.adoc (hunk 1 above)
Comment (ctargett) [reply solr/solr-ref-guide/modules/deployment-guide/pages/distributed-tracing.adoc:38]: Yes, you got the gist of what I meant correctly.

One of the challenges of including notes about behavior in specific versions is that at some point it becomes obsolete. A reader of the 9.4 Ref Guide may find it important to know that this feature is added in 9.4, but does the reader of 9.6? 10.0? One needs to remove such comments at some point, which turns it into a maintenance challenge.

Additionally, what is the value of telling someone in 9.4 that Solr 10 is going to include a different implementation? This comment simply mentions it in passing - does it impact how they implement in 9.4? If not, it should be left for the 10 release notes.

File: solr/solrj/src/java/org/apache/solr/common/params/CommonParams.java (hunk 2 above)
Comment (stillalex) [reply solr/solrj/src/java/org/apache/solr/common/params/CommonParams.java:297]: makes sense to not have this in the CommonParams. will move to the SearchHandler

**Comment by stillalex** [comment]
I marked `rid` as deprecated, but I can't find a good place in the guide to mention this deprecation. where should this go? under 'major changes in solr 9' maybe?

File: solr/solr-ref-guide/modules/deployment-guide/pages/distributed-tracing.adoc (hunk 1 above)
Comment (stillalex) [reply solr/solr-ref-guide/modules/deployment-guide/pages/distributed-tracing.adoc]: paragraph updated. please let me know if it looks better

**Comment by dsmiley** [comment]
Yes; definitely `major-changes-in-solr-9.adoc`

**Comment by stillalex** [comment]
updated the major changes doc. @dsmiley would appreciate a quick sanity check before merging.

**Review by dsmiley — APPROVED** [review]

**Pull request merged**